src/Compilers/CSharp/Test/Semantic/Semantics/UninitializedNonNullableFieldTests.cs

```diff
@@ -244,44 +287,167 @@ public void StaticFields_DefaultConstructor()
+                //     static C()
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "C").WithArguments("field", "F1").WithLocation(16, 12));
+        }
+
+        [Fact]
+        [WorkItem(34668, "https://github.com/dotnet/roslyn/issues/34668")]
+        [WorkItem(37511, "https://github.com/dotnet/roslyn/issues/37511")]
+        public void StaticFields_GenericTypes()
+        {
+            var source =
+@"#pragma warning disable 0169
+using System.Diagnostics.CodeAnalysis;
+class C<T, U, V>
+    where U : class
+    where V : struct
+{
+    private static T P1 { get; }
+    private static T P2 { get; } = default!;
+    [MaybeNull] private static T P3 { get; }
+    [MaybeNull] private static T P4 { get; } = default;
+    private static U P5 { get; set; }
+    private static U P6 { get; set; } = default!;
+    private static U? P7 { get; }
+    private static U? P8 { get; } = default;
+    private static V P9 { get; set; }
+    private static V P10 { get; set; } = default;
+    private static V? P11 { get; }
+    private static V? P12 { get; } = default;
+}";
+            var comp = CreateCompilation(new[] { MaybeNullAttributeDefinition, source }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (7,22): warning CS8618: Non-nullable property 'P1' is uninitialized. Consider declaring the property as nullable.
+                //     private static T P1 { get; }
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "P1").WithArguments("property", "P1").WithLocation(7, 22),
+                // (9,34): warning CS8618: Non-nullable property 'P3' is uninitialized. Consider declaring the property as nullable.
+                //     [MaybeNull] private static T P3 { get; }
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "P3").WithArguments("property", "P3").WithLocation(9, 34),
+                // (10,48): warning CS8653: A default expression introduces a null value when 'T' is a non-nullable reference type.
+                //     [MaybeNull] private static T P4 { get; } = default;
+                Diagnostic(ErrorCode.WRN_DefaultExpressionMayIntroduceNullT, "default").WithArguments("T").WithLocation(10, 48),
+                // (11,22): warning CS8618: Non-nullable property 'P5' is uninitialized. Consider declaring the property as nullable.
+                //     private static U P5 { get; set; }
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "P5").WithArguments("property", "P5").WithLocation(11, 22));
+        }
+
+        [Fact]
+        [WorkItem(34668, "https://github.com/dotnet/roslyn/issues/34668")]
+        public void StaticEvents()
+        {
+            var source =
+@"#pragma warning disable 0067
+#pragma warning disable 0414
+delegate void D();
+class C
+{
+    private static event D E1;
+    private static event D E2 = null!;
+    private static event D E3;
+    private static event D E4;
+    static C()
```

Comment:
>static C() [](start = 4, length = 10)

Consider test with static event but no cctor

Reply:
Consider testing `static event Action<int>? ...`

---
In reply to: [307936351](https://github.com/dotnet/roslyn/pull/37507#discussion_r307936351) [](ancestors = 307936351)

Reply:
Consider testing a nested type scenario (field/property/event)

---
In reply to: [307937002](https://github.com/dotnet/roslyn/pull/37507#discussion_r307937002) [](ancestors = 307937002,307936351)